app/src/main/cpp/skyline/services/audout/audout.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <audio.h>
+#include <services/base_service.h>
+#include <services/serviceman.h>
+#include <kernel/types/KEvent.h>
+
+namespace skyline::service::audout {
+    /**
+     * @brief audoutU is used to manage audio outputs
+     */
+    class audoutU : public BaseService {
+      public:
+        audoutU(const DeviceState &state, ServiceManager &manager);
+
+        /**
+         * @brief Returns a list of all available audio outputs
+         */
+        void ListAudioOuts(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Creates a new audoutU::IAudioOut object and returns a handle to it
+         */
+        void OpenAudioOut(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+    };
+
+    /**
+     * @brief audout:IAudioOut is a service opened when OpenAudioOut is called by audout
+     */
+    class IAudioOut : public BaseService {
+      public:
+        /**
+         * @param channelCount The channel count of the audio data the audio out will be fed
+         * @param sampleRate The sample rate of the audio data the audio out will be fed
+         */
+        IAudioOut(const DeviceState &state, ServiceManager &manager, int channelCount, int sampleRate);
+
+        std::shared_ptr<audio::AudioTrack> track; //!< The audio track associated with the audio out
+        std::shared_ptr<type::KEvent> releaseEvent; //!< The KEvent that is signalled when a buffer has been released
+        std::vector<i16> tmpSampleBuffer; //!< A temporary buffer used to store sample data in AppendAudioOutBuffer
+
+        /**
+         * @brief Returns the playback state of the audio output
+         */
+        void GetAudioOutState(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+        * @brief Starts playback using data from appended samples
+        */
+        void StartAudioOut(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+        * @brief Stops playback of audio, waits for all samples to be released
+        */
+        void StopAudioOut(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+        * @brief Appends sample data to the output buffer
+        */
+        void AppendAudioOutBuffer(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Returns a handle to the sample release KEvent
+         */
+        void RegisterBufferEvent(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Returns the IDs of all pending released buffers
+         */
+        void GetReleasedAudioOutBuffer(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Checks if the given buffer ID is in the playback queue
```

Comment:
Link to Switchbrew from the function: "... (https://switchbrew.org/wiki/Audio_services#ContainsAudioOutBuffer)"